deploy/podman/flightctl-api/flightctl-api-config/create_aap_application.sh

```
@@ -58,14 +58,18 @@ EOF
-    echo "Updating service config file with new client id: $client_id"
-    # Tempfile here is necessary because we cannot directly modify the mounted config file in place
-    tmpfile=$(mktemp)
-    trap 'rm -f "$tmpfile" "$oauth_output_file"' EXIT
-    if ! sed -E "s|^([[:space:]]*oAuthApplicationClientId:).*|\1 $client_id|" "$SERVICE_CONFIG_FILE" > "$tmpfile"; then
-        echo "Error: Failed to update config file" >&2
+    echo "Saving AAP OAuth client_id: $client_id"
+    # Write client_id to persistent file
+    # /certs is mounted from /etc/flightctl/pki on the host
+    client_id_file="/certs/aap-client-id"
+    echo "$client_id" > "$client_id_file"
+    if [ $? -eq 0 ]; then
```

Shellcheck (0.11.0)

[warning] This $? refers to echo/printf, not a previous command. Assign to variable to avoid it being overwritten.

(SC2320)